### ISSUE TITLE
tiled snap needs an update

### ISSUE THREAD

**Issue by pixelr0** [bjorn/tiled#2828]
tiled snap needs an update
https://snapcraft.io/tiled

snap version is 1.3.1
latest stable tiled version is 1.3.5

**Comment by bjorn** [comment]
Yeah, I'm relying on https://build.snapcraft.io/user/bjorn/tiled to build the snap and it's not clear to me how I could build anything else than the latest `master` branch. This is a problem for patch releases currently, since they are on a different branch (`1.3` in this case).

It means that I need to do a manual build of the snap for patch releases, which is a slow and error-prone process and I can only do it for amd64 architecture. Right now I'd rather just focus on getting Tiled 1.4.0 out, which should happen later today, and then the snap will be up-to-date again.

Long-term, one of these things should happen:

* `build.snapcraft.io` to support multiple branches (or I just need to figure out how that works, if it's already possible).
* Automate the manual process of building the snap, and have it run on the CI system, ideally for multiple architectures.
* Switch to a workflow where releases are always tagged on `master`. I'm considering this to reduce the maintenance overhead of two versions (though it would mean there is no more "stable" version of Tiled, but always just the "latest").

For now, please help making sure the 1.4.0 snap works fine by testing the build on the "edge" channel if you can.

**Comment by pixelr0** [comment]
sure :D i can test the 1.4.0 snap

**Comment by predr** [comment]
> Automate the manual process of building the snap, and have it run on the CI system, ideally for multiple architectures.

Checkout [action-publish](https://github.com/snapcore/action-publish). Sounds like it would fit your workflow needs.

**Comment by bjorn** [comment]
@predr Thanks, I'll definitely check that out!

**Comment by bjorn** [comment]
Snap was updated to 1.4.0 on Saturday. This release did come with a number of issues though, so a 1.4.1 release is planned as soon as possible (now probably tomorrow).

**Comment by hoxu** [comment]
Snap release seem to be lagging behind latest tag (v1.4.2) again:

```
$ snap info tiled |grep latest/stable
  latest/stable:    1.4.1               2020-06-25 (1825) 94MB -
```

Perhaps this issue could be renamed/edited to be about building snaps automatically by CI for each release.

It would be nice to have either up-to-date snaps or flatpaks on Linux, and it seems like [Flathub's org.mapeditor.Tiled](https://flathub.org/apps/details/org.mapeditor.Tiled) is even more outdated.

**Comment by pixelr0** [comment]
https://snapcraft.io/build

**Comment by bjorn** [comment]
> Perhaps this issue could be renamed/edited to be about building snaps automatically by CI for each release.

I guess it doesn't really matter what the issue is named. The reason it's still open is because the general issue is unresolved.

> It would be nice to have either up-to-date snaps or flatpaks on Linux, and it seems like [Flathub's org.mapeditor.Tiled](https://flathub.org/apps/details/org.mapeditor.Tiled) is even more outdated.

I've started to add the snap to the packaging workflow on GitHub Actions in #2928, though I'm not sure if it can update anything but the `amd64` build right now, whereas snapcraft.io is building for many more architectures. In any case, if we can have updated release builds for just `amd64` then it is already an improvement.

As for an up-to-date snap, you'll always find the bleeding edge version built from the `master` branch in the `edge` channel! :-)

Btw, I'm not currently maintaining a Flatpak for Tiled, so for any request for updates there you'd have to contact its author or whoever else can update it. But I think apart from being out of date it also appears to have a critical issue (https://github.com/flathub/org.mapeditor.Tiled/issues/4) which would need to be looked into.

**Comment by bjorn** [comment]
Now that #2928 is merged, a little more stable snap (updated every few weeks) is available in the `beta` channel, which is equivalent to installing the "snapshot" version from itch.io.

However, this issue will remain open for now, until the workflow is extended to also push releases to the candidate / stable channel.

**Comment by bjorn** [comment]
I've added documentation for how to build any version of the snap to the wiki:

https://github.com/mapeditor/tiled/wiki/Making-Releases#additional-architectures-via-remote-build

This has allowed publishing Tiled 1.12.2 as snap for all supported architectures, even with the tag on a branch (`1.12` rather than `master`). With that I guess we can finally close this issue.
